app/src/sharedTest/java/org/oppia/app/profile/ProfileChooserFragmentTest.kt

```diff
@@ -0,0 +1,153 @@
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.isDisplayed
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import org.hamcrest.Matchers.not
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.app.R
+import org.oppia.domain.profile.ProfileTestHelper
+import org.oppia.app.recyclerview.RecyclerViewMatcher.Companion.atPositionOnView
+import org.oppia.util.logging.EnableConsoleLog
+import org.oppia.util.logging.EnableFileLog
+import org.oppia.util.logging.GlobalLogLevel
+import org.oppia.util.logging.LogLevel
+import org.oppia.util.threading.BackgroundDispatcher
+import org.oppia.util.threading.BlockingDispatcher
+import javax.inject.Inject
+import javax.inject.Qualifier
+import javax.inject.Singleton
+
+@RunWith(AndroidJUnit4::class)
+class ProfileChooserFragmentTest {
+
+  @Inject lateinit var profileTestHelper: ProfileTestHelper
+  @Inject lateinit var context: Context
+
+  @Before
+  @ExperimentalCoroutinesApi
+  fun setUp() {
+    setUpTestApplicationComponent()
+    GlobalScope.launch(Dispatchers.Main) {
+      profileTestHelper.initializeProfiles()
+    }
+  }
+
+  private fun setUpTestApplicationComponent() {
+    DaggerProfileChooserFragmentTest_TestApplicationComponent.builder()
+      .setApplication(ApplicationProvider.getApplicationContext())
+      .build()
+      .inject(this)
+  }
+
+  @Test
+  fun testProfileChooserFragment_initializeProfiles_checkProfilesAreShown() {
+    ActivityScenario.launch(ProfileActivity::class.java).use {
+      onView(atPositionOnView(R.id.profile_recycler_view, 0, R.id.profile_name_text)).check(matches(withText("Sean")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 0, R.id.profile_is_admin_text)).check(matches(withText(context.getString(R.string.profile_chooser_admin))))
+      onView(atPositionOnView(R.id.profile_recycler_view, 1, R.id.profile_name_text)).check(matches(withText("Ben")))
+      onView(atPositionOnView(R.id.profile_recycler_view, 1, R.id.profile_is_admin_text)).check(matches(not(isDisplayed())))
+      onView(atPositionOnView(R.id.profile_recycler_view, 2, R.id.add_profile_text)).check(matches(withText(context.getString(R.string.profile_chooser_add))))
+    }
+  }
+
+  @Test
+  @ExperimentalCoroutinesApi
+  fun testProfileChooserFragment_addManyProfiles_checkProfilesSortedAndNoAddProfile() {
```

Comment:
Test case is failing with following error:
<img width="953" alt="Screenshot 2019-11-13 at 12 27 06 PM" src="https://user-images.githubusercontent.com/9396084/68740396-f6517b80-0610-11ea-87cc-82510abf004b.png">

Reply:
This might be because of corouties too.

Reply:
Updated to not use coroutines anymore. Please let me know if the pass test for you now because it works for me locally.